scipy/signal/tests/test_signaltools.py

```diff
@@ -573,10 +573,7 @@ def test_basic(self):
 
     def test_none(self):
         # Ticket #1124. Ensure this does not segfault.
-        try:
-            signal.medfilt(None)
-        except:
-            pass
+        signal.medfilt(None)
```

Comment:
It raises

Reply:
Tests pass on my machine and CIs (except for an unrelated error). Do I need to run some special testing mode?

Reply:
For the failed example, can you replace `>>> print(res.x)` by `>>> res.x`, and copy-paste the output?

It's just that the doctest checker does not understand the way numpy arrays are printed. Not sure why the error surfaces with this PR though.

Reply:
Ok, please disregard, I apparently misremember...

Reply:
@ev-br I made a commit to try what you asked and this was the output:
```
File "doc/source/tutorial/optimize.rst", line 699, in /home/travis/build/scipy/scipy/tools/../doc/source/tutorial/optimize.rst
Failed example:
    res.x
Expected:
    np.array([0.41494418 0.17011164])
Got:
    array([0.41494475, 0.1701105 ])
```
I'm just going to put a `# may vary` in there for now

Reply:
https://github.com/scipy/scipy/compare/master...ev-br:doctest_tutorial?expand=1 should fix it, I think.

Reply:
@ev-br and @larsoner I think the failure is due to #9131 which changed one of the input arguments, so the expected output changes also, it should be: `array([0.41494475, 0.1701105 ])` if you remove `print()`

Reply:
@mikofski can you push to #9138 or open a PR to fix it? I've reverted the changes here so we can address it separately.

Reply:
I've just placed a PR